upgrader.go

```diff
@@ -262,6 +262,20 @@ type neverCloseThisFile struct {
 
 func writePIDFile(path string) error {
 	dir, file := filepath.Split(path)
+
+	// if dir is empty, the user properly specified just the name
+	// of the pid file expecting it to be created in the current work directory
+	if dir == "" {
+		var err error
+		dir, err = os.Getwd()
+		if err != nil {
+			// if we could not determine current working directory
```

Comment:
I think it's safer to fail here. Any reason why this would not be acceptable?

Reply:
I added the `dir = ""` to make it more explicit to the reader what is happening and ensure backwards compatibility with the original code which continues and passes an empty dir variable to TempFile.

Do you happen to know the reasoning about this temporary file? - maybe we should just write the destination file out directly instead of moving it :thinking: 

That said - i think its totally acceptable to fail here - and i don't think anyone would ever trigger the fail (famous last words :)

Reply:
The temporary file is required to guarantee an atomic replacement of the pid file. Roughly, this: https://unix.stackexchange.com/a/35289